src/core/server/http/router/response_adapter.ts

```diff
@@ -120,7 +123,8 @@ export class HapiResponseAdapter {
     });
 
     error.output.payload.message = getErrorMessage(payload);
-    error.output.payload.attributes = getErrorAttributes(payload);
+    // Boom.output.payload is type limited to the default properties
+    (error.output.payload as any).attributes = getErrorAttributes(payload);
```

Comment:
This seems wrong? At least looking at @types/hapi__boom, I see an `attributes` property? https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/hapi__boom/index.d.ts#L87

Reply:
I don't think the payload is removed from the public API, the `API.md` file still reads the same:

> payload - the formatted object used as the response payload (stringified). Can be directly manipulated but any changes will be lost if reformat() is called. Any content allowed and by default includes the following content...

There's just no affordance for this in the types, which is probably for the best as you'd need some custom gate function to detect the type most likely, so I'm tempted to lead it as `(err.output.payload as any)`...

Reply:
Annoying, but ok this seems fine since these packages are all over the place, seems like the best we can do right now.

Reply:
1. `attributes` are not just payload. Boom adds them to `WWW-Authenticate` header https://github.com/hapijs/boom/blob/master/lib/index.js#L168-L202
2. probably we should update hapi typings. they do recommend to extend output payload  https://github.com/hapijs/boom/blob/master/API.md#faq

Reply:
Oh... I totally assumed attributes was referring to saved object attributes, and didn't realize it was a part of Boom at all 😬